File: app/src/main/java/com/smarttour/data/MapRepositoryImpl.kt
package com.smarttour.data

import com.smarttour.domain.MapRepository
import com.smarttour.domain.TourRoute
import com.smarttour.network.MapApiService

class MapRepositoryImpl(
    private val apiService: MapApiService
) : MapRepository {

    override suspend fun getActiveTourRoute(tourId: String): TourRoute {
        return apiService.getActiveTourRoute(tourId).toDomain()
    }

    override suspend fun markStopVisited(tourId: String, stopId: String) {
        apiService.markStopVisited(tourId, stopId)
    }
}

File: flutter_app/lib/data/repository/map_repository_impl.dart
import '../../domain/model/tour_route.dart';
import '../../domain/repository/map_repository.dart';
import '../api/map_api_service.dart';

/// Concrete implementation of [MapRepository] backed by [MapApiService].
class MapRepositoryImpl implements MapRepository {
  const MapRepositoryImpl(this._apiService);

  final MapApiService _apiService;

  @override
  Future<TourRoute> getActiveTourRoute(String tourId) =>
      _apiService.getActiveTourRoute(tourId);

  @override
  Future<void> markStopVisited(String tourId, String stopId) =>
      _apiService.markStopVisited(tourId, stopId);
}
